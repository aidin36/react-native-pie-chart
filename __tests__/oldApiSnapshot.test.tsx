import React from 'react'
import 'react-native'

import renderer from 'react-test-renderer'

import { it, expect } from '@jest/globals'

import PieChart from '../src/v3api'

it('five slices', () => {
  const series = [100, 200, 50]
  const colors = ['red', '#ff88bb', 'black']

  const rendered = renderer.create(<PieChart widthAndHeight={250} series={series} sliceColor={colors} />).toJSON()
  expect(rendered).toMatchSnapshot()
})

it('doughnut', () => {
  const series = [100, 200, 50]
  const colors = ['red', '#ff88bb', 'black']

  const rendered = renderer
    .create(<PieChart widthAndHeight={250} series={series} sliceColor={colors} colorRadius={0.3} />)
    .toJSON()
  expect(rendered).toMatchSnapshot()
})

it('doughnut with fill', () => {
  const series = [100, 200, 50]
  const colors = ['red', '#ff88bb', 'black']

  const rendered = renderer
    .create(<PieChart widthAndHeight={250} series={series} sliceColor={colors} colorRadius={0.3} colorFill='white' />)
    .toJSON()
  expect(rendered).toMatchSnapshot()
})

it('should throws error when length of color and series does not match', () => {
  const series = [100, 200, 50]
  const colors = ['red', '#ff88bb']

  expect(() => renderer.create(<PieChart widthAndHeight={250} series={series} sliceColor={colors} />)).toThrow(
    'Invalid "sliceColor": its length should be equal to the length of "series". sliceColor.length=2 series.length=3'
  )
})
