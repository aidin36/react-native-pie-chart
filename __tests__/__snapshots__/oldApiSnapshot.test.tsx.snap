// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doughnut 1`] = `
<RNSVGSvgView
  bbHeight={250}
  bbWidth={250}
  focusable={false}
  height={250}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {},
      {
        "flex": 0,
        "height": 250,
        "width": 250,
      },
    ]
  }
  width={250}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          1,
          0,
          0,
          1,
          125,
          125,
        ]
      }
    >
      <RNSVGPath
        d="M0,-125A125,125,0,0,1,121.866,27.815L0,0Z"
        fill={
          {
            "payload": 4294901760,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        d="M121.866,27.815A125,125,0,1,1,-97.729,-77.936L0,0Z"
        fill={
          {
            "payload": 4294936763,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        d="M-97.729,-77.936A125,125,0,0,1,0,-125L0,0Z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`five slices 1`] = `
<RNSVGSvgView
  bbHeight={250}
  bbWidth={250}
  focusable={false}
  height={250}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {},
      {
        "flex": 0,
        "height": 250,
        "width": 250,
      },
    ]
  }
  width={250}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          1,
          0,
          0,
          1,
          125,
          125,
        ]
      }
    >
      <RNSVGPath
        d="M0,-125A125,125,0,0,1,121.866,27.815L0,0Z"
        fill={
          {
            "payload": 4294901760,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        d="M121.866,27.815A125,125,0,1,1,-97.729,-77.936L0,0Z"
        fill={
          {
            "payload": 4294936763,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        d="M-97.729,-77.936A125,125,0,0,1,0,-125L0,0Z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGGroup>
</RNSVGSvgView>
`;
